File: 3.1. Binary Search/4. Find Minimum In Rotated Sorted Array/dory.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        left = 0
        right = len(nums) - 1

        while left < right:
            middle = (left + right) //2

            if nums[left] < nums[right]:
                return nums[left]
            elif nums[middle] >= nums[left]:
                left = middle +1
            else:
                right = middle
        return nums[left]

File: 4. Trees/1. Invert Binary Tree/dory.cs
# Runtime 0ms
# Beats 100.00%

# Memory 41.49MB
# Beats 72.67%

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 * public int val;
 * public TreeNode left;
 * public TreeNode right;
 * public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 * this.val = val;
 * this.left = left;
 * this.right = right;
 * }
 * }
 */
public class Solution {
    public TreeNode InvertTree(TreeNode root) {
        if(root == null) {
            return null;
        }
        TreeNode left = InvertTree(root.left);
        TreeNode right = InvertTree(root.right);
        root.left = right;
        root.right = left;

        return root;
    }
}